### PR TITLE
[user-authz] fix webhook crash on empty request data

### PR DESCRIPTION
**Pull request by libmonsoon-dev**
## Description

The termination of the process when receiving invalid data has been removed. 

## Why do we need it, and what problem does it solve?

(Close https://github.com/deckhouse/deckhouse/issues/5644)

## What is the expected result?

Now invalid data works the same as empty json

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section:140-user-authz
type: fix
summary: webhook no longer crashes if called without data
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've started the workflow for label 'e2e/run/static'...

**Review by yalosev — DISMISSED** [review]

File: ee/modules/140-user-authz/images/webhook/web/hook/handler.go [hunk 1]
```diff
@@ -74,8 +74,7 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	var request WebhookRequest
 	if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
-		// this case is exceptional
-		h.logger.Fatalf("cannot unmarshal kubernetes request: %v", err)
```
Comment (nabokihms) [inline ee/modules/140-user-authz/images/webhook/web/hook/handler.go:78]: Do we need to authorize an empty request? It seems like, at this point, we should return an error to a user with the `bad request` code.

File: ee/modules/140-user-authz/images/webhook/web/hook/handler.go [hunk 2]
```diff
@@ -74,8 +74,7 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+		h.logger.Printf("cannot unmarshal kubernetes request: %v", err)
```
Comment (yalosev) [inline ee/modules/140-user-authz/images/webhook/web/hook/handler.go:77]:
```suggestion
		http.Error(w, "Request body required", http.StatusBadRequest)
		return
```
somethink like that

File: ee/modules/140-user-authz/images/webhook/web/hook/handler.go (hunk 1 above)
Comment (libmonsoon-dev) [reply ee/modules/140-user-authz/images/webhook/web/hook/handler.go:78]: Done
image

**Pull request merged**